### PR TITLE
[SPARK-31090] Revert SPARK-25457 "IntegralDivide returns data type of the operands"

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This reverts commit 47d6e80a2e64823fabb596503fb6a6cc6f51f713.

### Why are the changes needed?

There is no standard requiring that `div` must return the type of the operand, and always returning long type looks fine. This is kind of a cosmetic change and we should avoid it if it breaks existing queries. This is similar to reverting TRIM function parameter order change.

### Does this PR introduce any user-facing change?

Yes, change the behavior of `div` back to be the same as 2.4.

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
@mgaido91 @maropu @dongjoon-hyun @HyukjinKwon

File: sql/core/src/test/resources/sql-tests/inputs/operators.sql [hunk 1]
```diff
@@ -22,6 +22,16 @@ select 5 / 0;
...
 
+-- integral div
+select 5 div 2;
+select 5 div 0;
+select 5 div null;
+select null div 5;
+select cast(51 as decimal(10, 0)) div cast(2 as decimal(2, 0));
+select cast(5 as decimal(1, 0)) div cast(0 as decimal(2, 0));
+select cast(5 as decimal(1, 0)) div cast(null as decimal(2, 0));
+select cast(null as decimal(1, 0)) div cast(5 as decimal(2, 0));
```
Comment (maropu) [inline sql/core/src/test/resources/sql-tests/inputs/operators.sql]: Were the four tests above newly added in this pr?

**Review by maropu — APPROVED** [review]
LGTM except for one minor comment.

**Review by dongjoon-hyun — COMMENTED** [review]
It seems that `postgreSQL/numeric.sql` fails. Could you take a look?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please hold on until the vote.

File: sql/core/src/test/resources/sql-tests/inputs/operators.sql (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/operators.sql]: They were already there. They were added when we support decimal in `IntegralDivide`

**Comment by dongjoon-hyun** [comment]
cc @rxin since he is the release manager for 3.0.0 (Also, cc @gatorsmile )

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Pull request closed without merging**